src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -51,54 +71,108 @@ export class ResultComponent implements OnInit, OnChanges {
     ) {}
 
     ngOnInit(): void {
-        if (this.result) {
+        // Get results initially if necessary
+        if (this.participation && !this.hasParticipationResults() && this.course && this.participation.exercise) {
+            this.resultService.findResultsForParticipation(this.course.id, this.participation.exercise.id, this.participation.id).subscribe(results => {
```

Comment:
No error handling. What happens if the call fails? `this.init()` would never be called.